index.js

```diff
@@ -2063,6 +2063,38 @@ export async function createNodejsBom(path, options) {
   let parentComponent = {};
   const parentSubComponents = [];
   let ppurl = "";
+  // Install deps for npm
+  const pkgJsonLockFile = getAllFiles(path, "package-lock.json", options);
+  const pkgJsonFile = getAllFiles(path, "package.json", options);
+  if (
+    pkgJsonLockFile?.length === 0 &&
+    pkgJsonFile?.length === 1 &&
+    options.installDeps
```

Comment:
This section must get moved down to line https://github.com/CycloneDX/cdxgen/blob/f58e80ecf7468949aa0b7c095d05bcf8456a8c3d/index.js#L2242 just above we iterate package lock files. Otherwise, it would always execute for every project including container.

Since the npm install would create new lock files, we need to repeat

```
pkgLockFiles = getAllFiles(
    path,
    `${options.multiProject ? "**/" : ""}package-lock.json`,
    options,
  );
```

Thank you in advance!